src/uzbl-ext.c

```diff
@@ -167,3 +189,60 @@ dom_blur_callback (WebKitDOMEventTarget *target,
     uzbl_extio_send_new_message (g_io_stream_get_output_stream (ext->stream),
                                  EXT_BLUR, name);
 }
+
+void
+ext_scroll_web_page (UzblExt      *ext,
+                     ScrollAxis    axis,
+                     ScrollAction  action,
+                     ScrollMode    mode,
+                     gint32       amount)
+{
+    WebKitDOMDocument *doc = webkit_web_page_get_dom_document (ext->web_page);
+    WebKitDOMDOMWindow *win = webkit_dom_document_get_default_view (doc);
+    WebKitDOMElement *body = WEBKIT_DOM_ELEMENT (
+        webkit_dom_document_get_body (doc));
+
+    if (action == SCROLL_BEGIN) {
+        webkit_dom_dom_window_scroll_to (
+            win,
+            axis == SCROLL_HORIZONTAL ? 0
+                                      : webkit_dom_dom_window_get_scroll_x (win),
+            axis == SCROLL_HORIZONTAL ? webkit_dom_dom_window_get_scroll_y (win)
+                                      : 0
+        );
+    } else if (action == SCROLL_END) {
+        webkit_dom_dom_window_scroll_to (
+            win,
+            axis == SCROLL_HORIZONTAL ? webkit_dom_element_get_scroll_width (body)
+                                      : webkit_dom_dom_window_get_scroll_x (win),
+            axis == SCROLL_HORIZONTAL ? webkit_dom_dom_window_get_scroll_y (win)
+                                      : webkit_dom_element_get_scroll_height (body)
+        );
+    } else if (action == SCROLL_SET) {
+        if (mode == SCROLL_PERCENTAGE) {
+            glong page = (axis == SCROLL_HORIZONTAL) ? webkit_dom_element_get_scroll_width (body)
+                                                     : webkit_dom_element_get_scroll_height (body);
+            amount = (gint32) page * amount * 0.01;
```

Comment:
Ah, yeah. The viewport size is what is missing then. Otherwise the range is discontinuous and there are some high percentages which are indistinguishable from each other.